### PR TITLE
Patient Zero - Midround Event

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Rare midround game-ending event: Patient Zero.
Hurls a solo romerol zombie at the station.

## Why It's Good For The Game

* People like zombies
* Zombies are not very common
* Zombies are easy to deal with and quite easy to manage
* Not such a threat lategame, if people are organized, but otherwise can be devastating
* Not as serious as benos/blob, but still threatening

## Changelog
:cl:
add: Patient zero mid-round event
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
Is it smart to set the minimum to 10?

**Comment by eeSPee** [comment]
Subject to change

File: code/modules/events/patient_zero.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	Mind.assigned_role = "Patient Zero"
+	Mind.special_role = "Patient Zero"
+	Mind.active = TRUE
+	Mind.transfer_to(patient)
+	patient.set_species(/datum/species/zombie/infectious)
+	patient.equipOutfit(/datum/outfit/pzero)
+	patient.dna.update_dna_identity()
+	
+	var/turf/T = locate(125,125,2)
+	patient.throw_at(T, 2, 2)
```
Comment (park66665) [inline code/modules/events/patient_zero.dm]:
```suggestion
	patient.throw_at(SSmapping.get_station_center(), 2, 2)
```

**Comment by wraith-54321** [comment]
the problem with zombies is people WILL try to get infected

File: code/modules/events/patient_zero.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	// 50% chance of being incremented by one
+	var/spawncount = 1
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/patient_zero/announce(fake)
+	priority_announce("Unidentified lifesigns detected coming aboard [station_name()]. Secure any exterior access, including ducting and ventilation.", "Lifesign Alert", 'sound/ai/aliens.ogg')
+
+/datum/round_event/ghost_role/patient_zero/spawn_role()
+	var/list/candidates = get_candidates(ROLE_ALIEN, null, ROLE_ALIEN)
```
Comment (PowerfulBacon) [inline code/modules/events/patient_zero.dm:19]: Role alien? There is one for midround antags

File: code/modules/events/patient_zero.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	minimum_required = 1
+	role_name = "patient zero"
+
+	// 50% chance of being incremented by one
+	var/spawncount = 1
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/patient_zero/announce(fake)
+	priority_announce("Unidentified lifesigns detected coming aboard [station_name()]. Secure any exterior access, including ducting and ventilation.", "Lifesign Alert", 'sound/ai/aliens.ogg')
```
Comment (PowerfulBacon) [inline code/modules/events/patient_zero.dm:16]: Probably should be unique considering it spawns an infectious zombie and is basically round ending

File: code/modules/events/patient_zero.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
+/datum/round_event_control/patient_zero
+	name = "Patient Zero"
+	typepath = /datum/round_event/ghost_role/patient_zero
+	weight = 5
```
Comment (PowerfulBacon) [inline code/modules/events/patient_zero.dm:4]: should definitely have a minimum time for a round ending event.

File: code/modules/events/patient_zero.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	min_players = 10
+
+/datum/round_event/ghost_role/patient_zero
+	announceWhen	= 400
+
+	minimum_required = 1
+	role_name = "patient zero"
+
+	// 50% chance of being incremented by one
+	var/spawncount = 1
```
Comment (PowerfulBacon) [inline code/modules/events/patient_zero.dm]: is this used?

File: code/modules/events/patient_zero.dm [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+/datum/round_event/ghost_role/patient_zero/spawn_role()
+	var/list/candidates = get_candidates(ROLE_ALIEN, null, ROLE_ALIEN)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected = pick(candidates)
+
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
```
Comment (PowerfulBacon) [inline code/modules/events/patient_zero.dm:26]: You are spawning them in space, wouldn't maints make more sense?

**Comment by DatBoiTim** [comment]
>Zombies are easy to deal with and quite easy to manage
 
Debatable. I've seen zombies trash the station even when advanced preparations were made.

Also regarding @PowerfulBacon 's comment regarding announcement: Meta-able announcements are lame. Especially since this is one of the few announcements that are still generic while pertaining to a handful of mobs. Also Xenomorphs can spawn with this announcement to my understanding, and xenomorphs are perhaps more dangerous than zombies.

File: code/modules/events/patient_zero.dm (hunk 6 above)
Comment (eeSPee) [reply code/modules/events/patient_zero.dm:26]: I see it as just a zombie floating away through space, which was lost (or thrown away) from another space ship/station. Hence the space spawn/being thrown at the station...

Plus it's a little bit different and it just doesn't appear out of nowhere in maints and needs to crawl its way into the station.

File: code/modules/events/patient_zero.dm (hunk 3 above)
Comment (eeSPee) [reply code/modules/events/patient_zero.dm:16]: I'll leave this to debate.

**Comment by francinum** [comment]
No.

**Comment by francinum** [comment]
>People like zombies

Debatable.
>Zombies are not very common

That's a good thing.
>Zombies are easy to deal with and quite easy to manage

*Debatable.*
>Not such a threat lategame, if people are organized, but otherwise can be devastating

No?
>Not as serious as benos/blob, but still threatening

what

**Comment by francinum** [comment]
Romerol is 25TC. Pay the tax if you want to hijack the round to zombies.

**Comment by ike709** [comment]
🔑

**Comment by TeomanTheGreat** [comment]
> >People like zombies
> 
> Debatable.
> >Zombies are not very common
> 
> That's a good thing.
> >Zombies are easy to deal with and quite easy to manage
> 
> *Debatable.*
> >Not such a threat lategame, if people are organized, but otherwise can be devastating
> 
> No?
> >Not as serious as benos/blob, but still threatening
> 
> what

Fran I've seen organized crews absolutely demolish zombies. I'm not sure where your getting the notion that this isn't possible. All you need is a sharp weapon for cutting their heads off and a group of doctors removing romerol tumors.

**Comment by wraith-54321** [comment]
thing is that only works when the people with guns dont run into maint and then get bit without telling anyone

**Review by eeSPee — COMMENTED** [review]
Disregard

**Comment by eeSPee** [comment]
12 updoot
6 herts

Closed without debate.

**Comment by OceanFish1** [comment]
> > > People like zombies
> > 
> > 
> > Debatable.
> > > Zombies are not very common
> > 
> > 
> > That's a good thing.
> > > Zombies are easy to deal with and quite easy to manage
> > 
> > 
> > _Debatable._
> > > Not such a threat lategame, if people are organized, but otherwise can be devastating
> > 
> > 
> > No?
> > > Not as serious as benos/blob, but still threatening
> > 
> > 
> > what
> 
> Fran I've seen organized crews absolutely demolish zombies. I'm not sure where your getting the notion that this isn't possible. All you need is a sharp weapon for cutting their heads off and a group of doctors removing romerol tumors.

No fun allowed lol cope harder kid

**Pull request closed without merging**